File: tests/benchmark/corpus/ruby/safe/safe_direct_path_sanitizer.rb
# ruby-safe-014: direct-return path sanitiser using `include?` / `start_with?`.
def sanitize_path(s)
  return '' if s.include?('..') || s.start_with?('/') || s.start_with?('\\')
  s
end

def handle(user_path)
  safe = sanitize_path(user_path)
  File.read(safe)
end

File: tests/benchmark/corpus/c/safe/safe_direct_path_sanitizer.c
/* c-safe-014: direct-return path sanitiser using strstr. */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

const char *sanitize_path(const char *s) {
    if (strstr(s, "..") != NULL || s[0] == '/' || s[0] == '\\') {
        return "";
    }
    return s;
}

void handle(const char *user_path) {
    const char *safe = sanitize_path(user_path);
    FILE *f = fopen(safe, "r");
    if (f) fclose(f);
}
